src/main/java/net/snowflake/client/core/arrow/tostringhelpers/ArrowArrayStringRepresentationBuilder.java

```diff
@@ -0,0 +1,20 @@
+package net.snowflake.client.core.arrow.tostringhelpers;
+
+import net.snowflake.client.core.SnowflakeJdbcInternalApi;
+import net.snowflake.client.jdbc.SnowflakeType;
+
+@SnowflakeJdbcInternalApi
+public class ArrowArrayStringRepresentationBuilder extends ArrowStringRepresentationBuilderBase {
+
+  private final SnowflakeType valueType;
+
+  public ArrowArrayStringRepresentationBuilder(SnowflakeType valueType) {
+    super("[", "]");
+    this.valueType = valueType;
+  }
+
+  public ArrowStringRepresentationBuilderBase appendValue(String value) {
+    addCommaIfNeeded();
```

Comment:
Did you weigh the pros and cons of using StringJoiner?

Reply:
I only considered two extreme cases of building the string manually (chosen) and using some abstraction like `JSONObject` (rejected) but didn't consider StringJointer which is an option in between so I'll also take a look at it